cvmfs/s3fanout.cc

```diff
@@ -636,8 +637,9 @@ int S3FanoutManager::InitializeDnsSettings(
   }
 
   // Remove port number if such exists
-  if (!HasPrefix(host_with_port, "http://", false /*ignore_case*/))
-    host_with_port = "http://" + host_with_port;
+  if (!HasPrefix(host_with_port, "http://", false /*ignore_case*/) ||
```

Comment:
I think that should be `&&` instead of `||`  Or perhaps: if `!IsHttpUrl(host_with_port)`